### PR TITLE
Adds "Create GitHub Issue" button to feedback modal

### LINKED ISSUE #2901: Add an alternative “Create a GitHub issue” button to feedback modal
Our feedback modal currently submits all feedback to Feedback Fish but if people have GitHub accounts it would be cool to funnel them to our issues as an alternative path (also to hint at people that this is where a lot of work happens!)

Bonus points if we can pre-fill the GitHub issue with any message a site visitor has already started authoring in the message box!

Example: here’s the form in [the AsyncAPI docs](https://www.asyncapi.com/docs/concepts)

Feedback modal that includes a “Create an issue” option

Feedback modal source code: https://github.com/withastro/docs/blob/main/src/components/tutorial/FeedbackButton.astro

### PR DESCRIPTION
**Pull request by Aperezpe**
#### What kind of changes does this PR include?

- New or updated content

#### Description

- Closes #2901
- What does this PR change?
  - It adds a "Create Github Issue" Button in the Github modal in addition to the Submit feedback button.
  - Increases the feedback modal width to fit both buttons and content well
  - Adds styling to the github button to style and switch between light and dark theme colors
  - Adds styling to the the feedback buttons for responsiveness
  - Adds 7 JS lines to pre-fill the GitHub issue with any message a site visitor has already typed in the message box
- Did you change something visual? A before/after screenshot can be helpful.

### Before:

Screenshot2023_03_22_150706
Screenshot2023_03_22_151950

<div style="display:flex">
  
 
</div>

### After:
Screenshot2023_03_22_150831
Screenshot2023_03_22_152028

<div style="display:flex;align-items:top;">
  
  
</div>

### CODE REVIEW

**Review by ElianCodes — APPROVED** [review]
Good work! Big fan of this!

File: src/components/tutorial/FeedbackButton.astro [hunk 1]
```diff
@@ -140,6 +159,13 @@ const instanceId = Math.round(Math.random() * 10e9).toString(36);
 			openBtn.addEventListener('click', openModal);
 			closeBtn.addEventListener('click', closeModal);
 
+			// Pre-fills the GitHub issue with any message a site visitor has already typed in the message box
+			githubButton?.addEventListener('click', () => {
+				if (form.text.value) 
+					githubButton.href = `https://github.com/withastro/docs/issues/new?body=${form.text.value}`
```
Comment (TheOtterlord) [inline src/components/tutorial/FeedbackButton.astro:165]: Should the form input be encoded to escape characters like `&` and `=` that might appear in the body? `?body=death & taxes` becomes `?body=death%20%26%20taxes`

```suggestion
					githubButton.href = `https://github.com/withastro/docs/issues/new?body=${encodeURIComponent(form.text.value)}`
```

**Review by TheOtterlord — CHANGES_REQUESTED** [review]
Looks awesome! :partying_face:

Found a small bug with the URL generation. I've suggested a fix below

File: src/components/tutorial/FeedbackButton.astro (hunk 1 above)
Comment (Aperezpe) [reply src/components/tutorial/FeedbackButton.astro]: That's true! Thanks for the suggestion!

File: src/components/tutorial/FeedbackButton.astro [hunk 2]
```diff
@@ -87,7 +87,25 @@ const instanceId = Math.round(Math.random() * 10e9).toString(36);
...
... 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z" class="astro-EBYKGQOL"
```
Comment (delucis) [inline src/components/tutorial/FeedbackButton.astro:103]: This `astro-EBYKGQOL` class was auto-generated elsewhere, so we can remove it here:

```suggestion
								d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"
```

**Review by delucis — COMMENTED** [review]
This is fantastic. Thank you for picking up this issue @Aperezpe!

I’ve made two suggestions but then this should be good to go 🙌

**Review by delucis — APPROVED** [review]
Thanks @Aperezpe! Will wait for the CI checks to pass, but I think this looks good to merge 🥳

**Comment by Aperezpe** [comment]
Thank you all for the suggestions and feedback! This is my very first contribution to an open source project, but you made it pretty easy for me! I am looking forward to contributing more in the future!

**Comment by delucis** [comment]
That’s great to hear! Welcome to Team Docs @Aperezpe 🚀

**Pull request merged**